File: Homework/Duy_Truong/Lesson5/454. 4Sum II.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @param {number[]} nums3
 * @param {number[]} nums4
 * @return {number}
 */
var fourSumCount = function(nums1, nums2, nums3, nums4) {
    let map = new Map; let count = 0
    for(let i=0;i<nums1.length;i++) {
        for(let j=0;j<nums2.length;j++) {
            let sum = nums1[i] + nums2[j]
            if(!map.has(sum))
                map.set(sum, 1)
            else
                map.set(sum, map.get(sum) + 1)
        }
    }
    for(let i=0;i<nums3.length;i++) {
        for(let j=0;j<nums4.length;j++) {
            if(map.get(-(nums3[i] + nums4[j])))
                count += map.get(-(nums3[i] + nums4[j]))
        }
    }
    
    return count
};

File: Homework/Ngan_Luu/454-4sum-ii.py
#https://leetcode.com/problems/4sum-ii/
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        count=collections.Counter()
        for x in nums1:
            for y in nums2:
                count[x+y]+=1
        output=0
        for x in nums3:
            for y in nums4:
                output+=count[0-x-y]
        return output